test_aodncore/pipeline/steps/test_upload.py

```diff
@@ -301,6 +301,49 @@ def test_invalid_bucket(self, mock_boto3):
 
         s3_upload_runner.s3_client.head_bucket.assert_called_with(Bucket=dummy_bucket)
 
+    @mock.patch('aodncore.pipeline.steps.upload.boto3')
+    def test_set_is_overwrite_not_present_s3(self, mock_boto3):
+        collection = get_upload_collection()
+        dummy_bucket = str(uuid4())
+        dummy_prefix = str(uuid4())
+        s3_upload_runner = S3UploadRunner(dummy_bucket, dummy_prefix, None, self.test_logger)
+        s3_upload_runner.set_is_overwrite(collection)
+        self.assertEqual(s3_upload_runner.s3_client.list_objects_v2.call_count, 4)
+        for pf in collection:
```

Comment:
Similar to below, this can be made more "functional" to kill the loop and multiple assertions:

```python
self.assertFalse(any(f.is_overwrite for f in collection))
```